x/exchange/keeper/commitments.go

```diff
@@ -426,3 +457,43 @@ func (k Keeper) consumeCommitmentSettlementFee(ctx sdk.Context, req *exchange.Ms
 	antewrapper.ConsumeMsgFee(ctx, exchangeFees.ExchangeFees, req, "")
 	return nil
 }
+
+// validateTransferCommitments which validate TransferCommitments message request.
+func (k Keeper) validateTransferCommitments(
+	ctx sdk.Context,
+	store storetypes.KVStore,
+	req *exchange.MsgMarketTransferCommitmentsRequest,
+	account sdk.AccAddress,
+) error {
+	// Blocked address check
+	if k.bankKeeper.BlockedAddr(account) {
+		return fmt.Errorf("%s is not allowed to receive funds", account)
+	}
+	// Zero amount
+	if req.Amount.IsZero() {
+		return fmt.Errorf("account %s does not have any funds committed to market %d", account, req.CurrentMarketId)
+	}
+
+	// Negative amount
+	if req.Amount.IsAnyNegative() {
+		return fmt.Errorf("cannot transfer negative commitment amount %q for %s in market %d", req.Amount, req.Account, req.CurrentMarketId)
+	}
+
+	// Market validity
+	if err := validateMarketIsAcceptingCommitments(store, req.CurrentMarketId); err != nil {
+		return fmt.Errorf("current market %d is invalid: %w", req.CurrentMarketId, err)
+	}
```

Comment:
The current market doesn't need to be accepting commitments for this.

Use case: A market used to allow commitments, but has since disabled them and now they want to transfer the existing commitments to another market.